pkg/ottl/expression.go

```diff
@@ -245,6 +246,39 @@ func (g StandardFloatGetter[K]) Get(ctx context.Context, tCtx K) (float64, error
 	}
 }
 
+type FunctionGetter[K any] interface {
+	Get(ctx context.Context, iArgs Arguments) (Expr[K], error)
+}
+
+type StandardFunctionGetter[K any] struct {
+	fCtx FunctionContext
+	fact Factory[K]
+}
+
+func (g StandardFunctionGetter[K]) Get(_ context.Context, iArgs Arguments) (Expr[K], error) {
+	if g.fact == nil {
+		return Expr[K]{}, fmt.Errorf("undefined function")
+	}
+	fArgs := g.fact.CreateDefaultArguments()
+	fArgsVal := reflect.ValueOf(fArgs).Elem()
+	iArgsVal := reflect.ValueOf(iArgs).Elem()
+
+	for i := 0; i < fArgsVal.NumField(); i++ {
+		field := iArgsVal.Field(i)
+		// The Argument passed in could be of any type, function Eval() will handle the type checking
+		fArgsVal.Field(i).Set(field)
```

Comment:
What will happen if `i` is out of range for  `iArgsVal`

Reply:
We should check up-front that `fArgs` and `iArgs` have the same length. We should also read in the indices for the fields from the `ottlargs` tag.

Reply:
ok I think we have this logic defined in `buildArgs` as well, I'll take another look.

Reply:
added these checks.